problem_builder/step.py

```diff
@@ -81,6 +83,13 @@ class MentoringStepBlock(
         scope=Scope.content
     )
 
+    # User state
+    student_results = List(
```

Comment:
@e-kolpakov Hm. Maybe for the odd case where a question (with a non-unique `name`) is used multiple times within the same step? With a `List` field we'd be able to retain results for all occurrences of the question in that case.